pkg/tangle/attacher.go

```diff
@@ -76,7 +77,7 @@ func (t *Tangle) MessageAttacher(opts ...MessageAttacherOption) *MessageAttacher
 	}
 }
 
-func (a *MessageAttacher) AttachMessage(ctx context.Context, msg *iotago.Message) (hornet.MessageID, error) {
+func (a *MessageAttacher) AttachMessage(ctx context.Context, msg *iotago.Message, powCompletedCallback PoWCompletedCallbackFunc) (hornet.MessageID, error) {
```

Comment:
why don't you pass a metrics object as MessageAttacherOption instead of adding a callback here?